### PR TITLE
Do not use BOOST_CURRENT_FUNCTION in MARK_REGION

### PR DESCRIPTION
**Pull request by aprokop**
BOOST_CURRENT_FUNCTION produces incredibly long names that make timer
output unreadable. The idea was good, but only if it printed just the
function name without templates or arguments.

### CODE REVIEW

**Comment by aprokop** [comment]
An example of what it used to be:
```|-> 2.21e+00 sec 25.8% 99.9% 0.0% 60 DTK:BVH:spatial [region]                                                                                                                                  
|   |-> 1.09e+00 sec 12.7% 100.0% 0.0% 60 DTK::BVH:second_pass [region]                                                                                                   
|   |   |-> 5.46e-01 sec 6.4% 100.0% 0.0% 30 [static void DataTransferKit::Details::BoundingVolumeHierarchyImpl<DeviceType>::queryDispatch(DataTransferKit::Details::SpatialPredicateTag, DataT
ransferKit::BoundingVolumeHierarchy<DeviceType>, Kokkos::View<Query*, DeviceType>, Kokkos::View<int*, DeviceType>&, Kokkos::View<int*, DeviceType>&, int) [with Query = DataTransferKit::Inters
ects<DataTransferKit::Sphere>; DeviceType = Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>]] second_pass [for]                                                                              
|   |   |-> 5.43e-01 sec 6.3% 100.0% 0.0% 30 [static void DataTransferKit::Details::BoundingVolumeHierarchyImpl<DeviceType>::queryDispatch(DataTransferKit::Details::SpatialPredicateTag, DataT
ransferKit::BoundingVolumeHierarchy<DeviceType>, Kokkos::View<Query*, DeviceType>, Kokkos::View<int*, DeviceType>&, Kokkos::View<int*, DeviceType>&, int) [with Query = DataTransferKit::Inters
ects<DataTransferKit::Sphere>; DeviceType = Kokkos::Device<Kokkos::Serial, Kokkos::HostSpace>]] second_pass [for]                                                                              
|   |-> 1.04e+00 sec 12.2% 100.0% 0.0% 60 DTK::BVH:first_pass [region]                                                                                                                         
|   |   |-> 5.25e-01 sec 6.1% 100.0% 0.0% 30 [static void DataTransferKit::Details::BoundingVolumeHierarchyImpl<DeviceType>::queryDispatch(DataTransferKit::Details::SpatialPredicateTag, DataT
ransferKit::BoundingVolumeHierarchy<DeviceType>, Kokkos::View<Query*, DeviceType>, Kokkos::View<int*, DeviceType>&, Kokkos::View<int*, DeviceType>&, int) [with Query = DataTransferKit::Inters
ects<DataTransferKit::Sphere>; DeviceType = Kokkos::Device<Kokkos::Serial, Kokkos::HostSpace>]] first_pass_at_the_search_count_the_number_of_indices [for]                                     
|   |   |-> 5.15e-01 sec 6.0% 100.0% 0.0% 30 [static void DataTransferKit::Details::BoundingVolumeHierarchyImpl<DeviceType>::queryDispatch(DataTransferKit::Details::SpatialPredicateTag, DataT
ransferKit::BoundingVolumeHierarchy<DeviceType>, Kokkos::View<Query*, DeviceType>, Kokkos::View<int*, DeviceType>&, Kokkos::View<int*, DeviceType>&, int) [with Query = DataTransferKit::Inters
ects<DataTransferKit::Sphere>; DeviceType = Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>]] first_pass_at_the_search_count_the_number_of_indices [for]    

```
and what it is now
```
|-> 7.71e-01 sec 19.4% 100.0% 0.0% 8 DTK_perform_nearest_queries [for]
|   |-> 7.71e-01 sec 19.4% 100.0% 0.0% 8 DTK:BVH:traversal [region]
|       |-> 7.71e-01 sec 19.4% 100.0% 0.0% 8 DTK:BVH:knn [region]             
|-> 5.91e-01 sec 14.9% 100.0% 0.0% 7 DTK_generate_hierarchy [for]        
|   |-> 5.91e-01 sec 14.9% 100.0% 0.0% 7 DTK:BVH:generate_hierarchy [region]
|       |-> 5.91e-01 sec 14.9% 100.0% 0.0% 7 DTK:BVH:construction [region]
|-> 4.14e-01 sec 10.4% 100.0% 0.0% 25 Kokkos::Sort::BinBinning [for]
|   |-> 4.05e-01 sec 10.2% 100.0% 0.0% 7 DTK:BVH:initialize_leaf_nodes [region]
|   |   |-> 4.05e-01 sec 10.2% 100.0% 0.0% 7 DTK:BVH:construction [region]
|   |-> 8.77e-03 sec 0.2% 100.0% 0.0% 18 DTK:BVH:sort_queries [region]
|       |-> 4.89e-03 sec 0.1% 100.0% 0.0% 10 DTK:BVH:spatial [region]
|-> 3.82e-01 sec 9.6% 100.0% 0.0% 7 DTK_initialize_leaf_nodes [for]
|   |-> 3.82e-01 sec 9.6% 100.0% 0.0% 7 DTK:BVH:initialize_leaf_nodes [region] 
```
```

File: packages/Utils/src/DTK_ConfigDefs.hpp [hunk 1]
```diff
@@ -30,13 +30,7 @@ namespace DataTransferKit
...
 #include "DTK_Types.h"
 
-// clang-format off
-#ifdef HAVE_DTK_BOOST
-#define DTK_MARK_REGION(x) std::string("[")+BOOST_CURRENT_FUNCTION+"] "+x
-#else
-#define DTK_MARK_REGION(x) x
-#endif
-// clang-format on
+#define DTK_MARK_REGION( x ) std::string( "DTK_" ) + x
```
Comment (dalg24) [inline packages/Utils/src/DTK_ConfigDefs.hpp]: No strong opinion on this.  This is fine by me.

**Comment by dalg24** [comment]
Remove https://github.com/ORNL-CEES/DataTransferKit/blob/4b2f2646c06db90042102aac5b1294b9dbab0733/cmake/TPLs/FindTPLBoostOrg.cmake#L12

**Comment by dalg24** [comment]
Great now we get sporadic timeouts on CUDA.  This also occurred on #535

**Comment by dalg24** [comment]
"The definition of insanity is doing the same thing over and over again and expecting a different result."

**Pull request merged**